scripts/reorder_chains.py

```
@@ -0,0 +1,450 @@
+    try:
+        from proxy_app.penalty_store import PenaltyStore  # type: ignore
+
+        store = PenaltyStore.get()
+        entries = store.get_entries(now=now)
+        scores: Dict[Tuple[str, str], float] = {}
+        for e in entries:
+            key = (e.provider, e.model)
+            # Sum decayed score across failure types (get_entries already decays).
+            scores[key] = scores.get(key, 0.0) + e.score
+        return scores
+    except Exception as exc:
```

Ruff (0.15.17)

[warning] Do not catch blind exception: `Exception`

(BLE001)